helper_test.go

```diff
@@ -1302,18 +1302,19 @@ func waitForRunLock(t *testing.T, client *Client, workspaceID string) {
 	wg.Wait()
 }
 
-func retry(f retryableFn) (interface{}, error) {
+func retry(f retryableFn) (interface{}, error) { //nolint
 	tick := time.NewTicker(tickDuration * time.Second)
-
 	retries := 0
 	maxRetries := 5
 
-	for {
+	defer tick.Stop()
+
+	for { //nolint
```

Comment:
We discussed the two approaches for this block. The linter would have us simplify this block to iterate over the range of the ticker values, but we prefer to stick with the for / select option because it's consistent with how we use tickers in this library.